docs/source/rag.rst

```diff
@@ -0,0 +1,99 @@
+.. _rag-documentation:
+
+RAG Documentation
+#################
+
+************************************
+Retrieval-Augmented Generation (RAG)
+************************************
+RAG is a technique that enhances Large Language Model (LLM) responses by retrieving
+source information from external data stores to augment generated responses.
+
+This package enables Python developers to perform RAG using Neo4j.
+
+
+************************************
+Overview
+************************************
+
+.. code:: python
+
+    from neo4j import GraphDatabase
+    from neo4j_genai import VectorRetriever, OpenAILLM, GraphRAG
+
+    URI = "neo4j://localhost:7687"
+    AUTH = ("neo4j", "password")
+
+    INDEX_NAME = "embedding-name"
+
+    # Connect to Neo4j database
+    driver = GraphDatabase.driver(URI, auth=AUTH)
+
+    # Create Embedder object
+    embedder = OpenAIEmbeddings(model="text-embedding-3-large")
+
+    # Initialize the retriever
+    retriever = VectorRetriever(driver, INDEX_NAME, embedder)
+
+    # Initialize the LLM
+    # Note: the OPENAI_API_KEY must be in the env vars
+    llm = OpenAILLM(model_name="gpt-4o", model_params={"temperature": 0})
+
+    # Initialize the RAG pipeline
+    rag = GraphRAG(retriever=vector_retriever, llm=llm)
+
+    # Query the graph
+    query_text = "How do I do similarity search in Neo4j?"
+    response = rag.search(query_text=query_text, retriever_config={"top_k": 5})
```

Comment:
I would add this example to the README as well, replacing the vector search that's there now.

Reply:
Makes sense, added in https://github.com/neo4j/neo4j-genai-python/pull/50/commits/73a05fd58cedbd961416ead2dc57b86726617639